clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -1006,6 +1036,9 @@ public String toEnrichedRst() {
 
         String lastKeyGroupName = "";
         for (ConfigKey key : sortedConfigs()) {
+            if (key.deprecated) {
+                continue;
```

Comment:
Are we just skipping deprecated configs? The approach in the following PR from @cmccabe seems better:

https://github.com/apache/kafka/pull/2456/files